visualizer/semanticsEditor.js

```diff
@@ -360,6 +366,177 @@
   }
   ohmEditor.parseTree.addListener('cmdclick:traceElement', toggleSemanticsEditor);
 
+  // Retrieve arguments from the menu entry's submenu
+  function retrieveArgumentsFromSubList(entry) {
+    var argumentWrappers = entry.querySelectorAll('ul li');
+    if (!argumentWrappers) {
+      return undefined;
+    }
+
+    var operationArguments = Object.create(null);
+    Array.prototype.forEach.call(argumentWrappers, function(argumentWrapper) {
+      var name = argumentWrapper.querySelector('.name').textContent;
+      var valueContainer = argumentWrapper.querySelector('.value');
+      var value = eval(valueContainer.value);    // eslint-disable-line no-eval
+      operationArguments[name] = value;
+    });
+    return operationArguments;
+  }
+
+  // Create a menu entry that wraps the argument name and a textarea for user to enter
+  // corresponding value.
+  function createArgumentWrapper(name) {
+    var argumentWrapper = domUtil.createElement('li');
+    argumentWrapper.appendChild(domUtil.createElement('.name', name));
+    argumentWrapper.appendChild(domUtil.createElement('.assign', UnicodeChars.LEFTWARDS_ARROW));
+    var valueContainer = argumentWrapper.appendChild(domUtil.createElement('textarea.value'));
+    valueContainer.placeholder = 'value';
+    valueContainer.cols = 5;
+    valueContainer.addEventListener('keyup', function(e) {
```

Comment:
I like to do `onkeyup = function(e) { ... }` if I know that there is only one listener. That way, you know that if someone needs to add a second listener for the same event, they will look at this code (and possibly combine the two).